client/trino-jdbc/src/test/java/io/trino/jdbc/BaseTestJdbcResultSet.java

```diff
@@ -283,24 +283,12 @@ public void testDate()
             // date which midnight does not exist in test JVM zone
             checkRepresentation(connectedStatement.getStatement(), "DATE '1970-01-01'", Types.DATE, (rs, column) -> {
                 LocalDate localDate = LocalDate.of(1970, 1, 1);
+                Date sqlDate = Date.valueOf(localDate);
 
-                // TODO (https://github.com/trinodb/trino/issues/6242) this should not fail
-                assertThatThrownBy(() -> rs.getObject(column))
-                        .isInstanceOf(SQLException.class)
-                        .hasMessage("Expected value to be a date but is: 1970-01-01")
-                        .hasStackTraceContaining("Cannot parse \"1970-01-01\": Illegal instant due to time zone offset transition (America/Bahia_Banderas)");
-                // TODO (https://github.com/trinodb/trino/issues/6242) this should not fail
-                assertThatThrownBy(() -> rs.getObject(column, Date.class))
-                        .isInstanceOf(SQLException.class)
-                        .hasMessage("Expected value to be a date but is: 1970-01-01")
-                        .hasStackTraceContaining("Cannot parse \"1970-01-01\": Illegal instant due to time zone offset transition (America/Bahia_Banderas)");
+                assertEquals(rs.getObject(column), sqlDate);
+                assertEquals(rs.getObject(column, Date.class), sqlDate);
                 // TODO assertEquals(rs.getObject(column, LocalDate.class), localDate);
-
-                // TODO (https://github.com/trinodb/trino/issues/6242) this should not fail
```

Comment:
Are you fixing https://github.com/trinodb/trino/issues/6242?

Reply:
Yes.